app/workers/email_archive_worker.rb

```diff
@@ -0,0 +1,85 @@
+class EmailArchiveWorker
+  include Sidekiq::Worker
+
+  sidekiq_options queue: :cleanup
+
+  LOCK_NAME = "email_archive_worker".freeze
+
+  def perform
+    Email.with_advisory_lock(LOCK_NAME, timeout_seconds: 0) do
+      start_time = Time.zone.now
+      archived = 0
+
+      EmailArchiveQuery.call.in_batches do |batch|
+        Email.transaction do
+          archived += archive_batch(batch)
```

Comment:
Might be tempted to call this `archived_count` as at first I thought this was storing each individual archived email.